test/testdrive/kafka-recreate-topic.td

```diff
@@ -108,11 +108,12 @@ text
 ----
 1
 
+# TODO(guswynn): ensure this error stays in place even when _environmentd_ restarts.
 > SELECT name, status, error FROM mz_internal.mz_source_statuses WHERE type != 'progress'
 name            status    error
 -------------------------------
 good_source     running   <null>
 source0         stalled  "kafka: topic was recreated: partition count regressed from 4 to 2"
 source1         stalled  "kafka: topic was recreated: high watermark of partition 0 regressed from 1 to 0"
 
-> DROP CLUSTER IF EXISTS to_recreate CASCADE;
+# > DROP CLUSTER IF EXISTS to_recreate CASCADE;
```

Comment:
This causes a failure in nightly:

```
linked-clusters.td
Connecting to PostgreSQL server at postgres://mz_system:materialize@materialized:6877...
> DROP DATABASE materialize
skip-if query returned false; continuing
> CREATE MATERIALIZED VIEW v AS SELECT 1 AS a
rows match; continuing at ts 1700122456.351414
> SELECT name FROM (SHOW CLUSTERS)
rows didn't match; sleeping to see if dataflow catches up 50ms 75ms 113ms 169ms 253ms 380ms 570ms 854ms 1s 2s 3s 4s 6s 10s 15s 22s 33s 22s
linked-clusters.td:20:1: error: non-matching rows: expected:
[["default"], ["mz_introspection"], ["mz_system"]]
got:
[["default"], ["mz_introspection"], ["mz_system"], ["to_recreate"]]
```

https://buildkite.com/materialize/nightlies/builds/5231#018bd70f-a074-403b-ac93-c10352f58c6e

Reply:
oh yeah this shouldnt have been committed! ill rerun nightlies after!